src/wallet/wallet.cpp

```diff
@@ -1018,9 +1018,7 @@ bool CWallet::AbandonTransaction(const uint256& hashTx)
             // Iterate over all its outputs, and mark transactions in the wallet that spend them abandoned too
             TxSpends::const_iterator iter = mapTxSpends.lower_bound(COutPoint(now, 0));
             while (iter != mapTxSpends.end() && iter->first.hash == now) {
-                if (!done.count(iter->second)) {
-                    todo.insert(iter->second);
-                }
+                todo.emplace(iter->second);
```

Comment:
I'm not sure if I understand this change, but the first check is in the set `done` not `todo` so it isn't obvious to me why this is correct

Reply:
I think the assumption is that sha256 hashes don't collide, but I agree the explanation/reasoning can be a bit more verbose